src/lib/list/list.ts

```diff
@@ -105,20 +94,19 @@ export class MdListIconCssMatStyler {}
  */
 @Directive({
   selector: '[md-subheader], [mat-subheader]',
-  host: {
-    '[class.mat-subheader]': 'true'
-  }
+  host: {'class': 'mat-subheader'}
 })
 export class MdListSubheaderCssMatStyler {}
 
 @Component({
   moduleId: module.id,
-  selector: 'md-list-item, mat-list-item, a[md-list-item], a[mat-list-item]',
+  selector: 'md-list-item, mat-list-item, a[md-list-item], a[mat-list-item],' +
+            'a[mdListItem], a[matListItem]',
```

Comment:
So it looks like we're just doing away with dash-case attrs entirely, even for components? (I'm happy with that, I've wanted it for a while) shouldn't we do button too?